### PR TITLE
fixes openarray hoist with gcc 14

### PR DESCRIPTION
**Pull request by ringabout**
blocks https://github.com/nim-lang/Nim/pull/23673

### CODE REVIEW

File: compiler/semexprs.nim [hunk 1]
```diff
@@ -2974,8 +2990,16 @@ proc hoistParamsUsedInDefault(c: PContext, call, letSection, defExpr: var PNode)
         newNodeI(nkEmpty, letSection.info),
         call[paramPos])
 
+      # echo " -> ", letSection.renderTree
+      # echo (call[paramPos].typ, hoistedVarSym.typ, call[paramPos].renderTree)
+      # debug call[paramPos]
+      # debug call[paramPos].typ
+      # echo isOpenArraySym
+
```
Comment (Araq) [inline compiler/semexprs.nim]:
```suggestion
```

File: compiler/semexprs.nim [hunk 2]
```diff
@@ -2974,8 +2990,16 @@ proc hoistParamsUsedInDefault(c: PContext, call, letSection, defExpr: var PNode)
...
       call[paramPos] = newSymNode(hoistedVarSym) # Refer the original arg to its hoisted sym
 
+      
+
```
Comment (Araq) [inline compiler/semexprs.nim]:
```suggestion
```

File: compiler/semexprs.nim [hunk 3]
```diff
@@ -2944,6 +2944,19 @@ proc asBracketExpr(c: PContext; n: PNode): PNode =
           return result
   return nil
 
+proc isOpenArraySym(x: PNode): bool =
+  result = false
```
Comment (Araq) [inline compiler/semexprs.nim]:
```suggestion
```

**Comment by Araq** [comment]
The test case hasn't been updated?

**Comment by ringabout** [comment]
Yeah, the exact test failed with gcc 14 so it doesn't need to be updated. 

Shown in https://github.com/nim-lang/Nim/pull/23673

> /home/runner/work/Nim/Nim/nimcache/tests/megatest_4a8a08f09d37b73795649038408b5f33/@mccgbugs@st10964.nim.c:96:35: error: assignment to ‘tyOpenArray__YZIefVt4yx7h5gisKZd4WQ *’ from incompatible pointer type ‘tySequence__48JTohSgTy339bRxHzUD8KA *’ [-Wincompatible-pointer-types]

**Comment by ringabout** [comment]
I tested it locally with updated changes, it worked with gcc 14

**Pull request merged**